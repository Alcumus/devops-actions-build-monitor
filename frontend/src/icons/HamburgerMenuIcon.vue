<template>
  <div
    class="icon-container"
    :class="{'opened':isClicked}"
    @click="openMenu"
  >
    <span />
    <span />
    <span />
  </div>
</template>

<script>
export default {
  name: 'HamburgerMenuIcon',
  props: {
    isClicked: Boolean
  },
  emits: ['clicked'],
  methods: {
    openMenu() {
      this.$emit('clicked');
    }
  }
};
</script>

<style scoped>
.icon-container {
  height: 24px;
  width: 24px;
  margin-bottom: 40px;
  margin-top: 10px;
  margin-left: 7px;
}

.icon-container:hover {
  cursor: pointer;
}

.icon-container span {
  height: 3px;
  width: 24px;
  display: inline-block;
  background-color: #fff;
  position: relative;
  border-radius: 5px;
  transition: 0.3s cubic-bezier(0.8, 0.5, 0.2, 1.4);
  transition-duration: 500ms;
}

.icon-container span:first-child {
  top: 10px;
}

.icon-container span:last-child {
  bottom: 10px;
}

.icon-container:not(.opened) {
  margin: 10px auto 40px;
}

.icon-container.opened span:first-child {
  transform: rotate(-45deg);
  top: 18.5px;
}

.icon-container.opened span:nth-child(2) {
  opacity: 0;
}

.icon-container.opened span:last-child {
  transform: rotate(45deg);
  bottom: 17px;
}

</style>
