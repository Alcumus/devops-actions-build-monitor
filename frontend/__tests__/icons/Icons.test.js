import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import InvalidIcon from '@/icons/InvalidIcon';
import LogoutIcon from '@/icons/LogoutIcon';
import CodeStandardViolationsIcon from '@/icons/CodeStandardViolationsIcon';
import SecretsIcon from '@/icons/SecretsIcon';
import SettingsIcon from '@/icons/SettingsIcon';
import WorkflowJobsIcon from '@/icons/WorkflowJobsIcon';
import HideOrShowIcon from '@/icons/HideOrShowIcon';
import HamburgerMenuIcon from '@/icons/HamburgerMenuIcon';

describe('Icons', () => {
  it.each([
    { component: InvalidIcon, name: 'InvalidIcon' },
    { component: LogoutIcon, name: 'LogoutIcon' },
    { component: CodeStandardViolationsIcon, name: 'CodeStandardViolationsIcon' },
    { component: SecretsIcon, name: 'SecretsIcon' },
    { component: SettingsIcon, name: 'SettingsIcon' },
    { component: WorkflowJobsIcon, name: 'WorkflowJobsIcon' }
  ])('should render $name', ({ component, name }) => {
    expect(mount(component).html()).toMatchSnapshot();
  });

  describe('HideOrShowIcon', () => {
    it('should render hide icon', () => {
      const wrapper = mount(HideOrShowIcon, {
        props: {
          displayHideButton: true
        }
      });
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('HamburgerMenuIcon', () => {
    it('should render hamburger menu icon in closed state', () => {
      const wrapper = mount(HamburgerMenuIcon, {
        props: {
          isClicked: false
        }
      });
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render hamburger menu icon in opened state', () => {
      const wrapper = mount(HamburgerMenuIcon, {
        props: {
          isClicked: true
        }
      });
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should emit clicked event', async () => {
      const wrapper = mount(HamburgerMenuIcon, {
        props: {
          isClicked: false
        }
      });

      await wrapper.find('.icon-container').trigger('click');
      expect(wrapper.emitted()).toHaveProperty('clicked');
    });
  });
});
