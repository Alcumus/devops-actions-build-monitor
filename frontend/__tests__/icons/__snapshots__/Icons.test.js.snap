// Vitest Snapshot v1

exports[`Icons > HamburgerMenuIcon > should render hamburger menu icon in closed state 1`] = `
<div class="icon-container">
  <span></span>
  <span></span>
  <span></span>
</div>
`;

exports[`Icons > HamburgerMenuIcon > should render hamburger menu icon in opened state 1`] = `
<div class="icon-container opened">
  <span></span>
  <span></span>
  <span></span>
</div>
`;

exports[`Icons > HideOrShowIcon > should render hide icon 1`] = `
<svg height="48" viewBox="0 0 48 48" width="48" xmlns="http://www.w3.org/2000/svg">
  <!--  eslint-disable max-len -->
  <path d="m40.8 44.8-8.4-8.25q-1.75.7-3.95 1.075T24 38q-7.3 0-13.25-4.075T2 23q1-2.6 2.775-5.075T9.1 13.2L2.8 6.9l2.1-2.15L42.75 42.6ZM24 31.5q.7 0 1.5-.125T26.85 31L16 20.15q-.25.6-.375 1.35-.125.75-.125 1.5 0 3.6 2.5 6.05 2.5 2.45 6 2.45Zm13.9 2-6.45-6.45q.5-.8.775-1.875T32.5 23q0-3.55-2.475-6.025Q27.55 14.5 24 14.5q-1.1 0-2.15.25-1.05.25-1.9.8L14.45 10q1.75-.8 4.475-1.4Q21.65 8 24.25 8q7.15 0 13.075 4.075Q43.25 16.15 46 23q-1.3 3.2-3.35 5.85-2.05 2.65-4.75 4.65Zm-8.65-8.65-7.1-7.1q1.45-.55 3-.225 1.55.325 2.7 1.425 1.15 1.15 1.6 2.575.45 1.425-.2 3.325Z" /><!--  eslint-enable max-len -->
</svg>
<!--v-if-->
`;

exports[`Icons > should render 'CodeStandardViolationsIcon' 1`] = `
<svg height="30" viewBox="0 0 256 256" width="30" xmlns="http://www.w3.org/2000/svg">
  <polyline fill="none" points="64 88 16 128 64 168" stroke="#fff" stroke-linecap="round" stroke-linejoin="round" stroke-width="24" />
  <polyline fill="none" points="192 88 240 128 192 168" stroke="#fff" stroke-linecap="round" stroke-linejoin="round" stroke-width="24" />
  <line fill="none" stroke="#fff" stroke-linecap="round" stroke-linejoin="round" stroke-width="24" x1="160" x2="96" y1="40" y2="216" />
</svg>
`;

exports[`Icons > should render 'InvalidIcon' 1`] = `
<svg id="Layer_1" style="enable-background: new 0 0 496 496;" viewBox="0 0 496 496" x="0px" xml:space="preserve" xmlns="http://www.w3.org/2000/svg" y="0px">
  <path d="M496,248c0,136.8-111.2,248-248,248S0,384.8,0,248S111.2,0,248,0S496,111.2,496,248z" style="fill:#b5092b;" />
  <path d="M248,0c136.8,0,248,111.2,248,248S384.8,496,248,496" style="fill:#890027;" />
  <path d="M72.8,72.8c96.8-96.8,253.6-96.8,350.4,0s96.8,253.6,0,350.4" style="fill:#9b0025;" />
  <g>
    <!--  eslint-disable no-tabs -->
    <path d="M321.6,333.6c-3.2,0-5.6-0.8-8-3.2l-148-148c-4.8-4.8-4.8-12,0-16.8s12-4.8,16.8,0l148,148
		c4.8,4.8,4.8,12,0,16.8C328,332.8,324.8,333.6,321.6,333.6z" style="fill:#eff;" />
    <path d="M174.4,333.6c-3.2,0-5.6-0.8-8-3.2c-4.8-4.8-4.8-12,0-16.8l148-148c4.8-4.8,12-4.8,16.8,0
		s4.8,12,0,16.8l-148.8,148C180,332.8,176.8,333.6,174.4,333.6z" style="fill:#eff;" /><!--  eslint-enable no-tabs -->
  </g>
  <g />
  <g />
  <g />
  <g />
  <g />
  <g />
  <g />
  <g />
  <g />
  <g />
  <g />
  <g />
  <g />
  <g />
  <g />
</svg>
`;

exports[`Icons > should render 'LogoutIcon' 1`] = `
<svg fill="#FFF" height="30" id="Icons" style="enable-background: new 0 0 32 32;" viewBox="0 0 32 32" width="30" x="0px" xml:space="preserve" xmlns="http://www.w3.org/2000/svg" y="0px">
  <!--  eslint-disable max-len no-tabs -->
  <path d="M30.9,13.6c-0.1-0.1-0.1-0.2-0.2-0.3l-4-4c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l2.3,2.3H22v-3V3c0-0.6-0.4-1-1-1H4
	c0,0,0,0,0,0C3.9,2,3.7,2,3.6,2.1c0,0,0,0-0.1,0c0,0-0.1,0-0.1,0.1c0,0-0.1,0.1-0.1,0.1c0,0,0,0,0,0C3.2,2.4,3.1,2.5,3.1,2.6
	c0,0,0,0,0,0.1C3,2.8,3,2.9,3,3v22c0,0.4,0.2,0.8,0.6,0.9l9,4C12.7,30,12.9,30,13,30c0.2,0,0.4-0.1,0.5-0.2c0.3-0.2,0.5-0.5,0.5-0.8
	v-3h7c0.6,0,1-0.4,1-1V15h5.6l-2.3,2.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l4-4c0.1-0.1,0.2-0.2,0.2-0.3
	C31,14.1,31,13.9,30.9,13.6z M10,21c0,0.6-0.4,1-1,1s-1-0.4-1-1v-4c0-0.6,0.4-1,1-1s1,0.4,1,1V21z M20,10v14h-6V7
	c0-0.4-0.2-0.8-0.6-0.9L8.7,4H20V10z" /><!--  eslint-enable max-len no-tabs -->
</svg>
`;

exports[`Icons > should render 'SecretsIcon' 1`] = `
<svg height="30" viewBox="0 0 512 512" width="30" xmlns="http://www.w3.org/2000/svg">
  <!--  eslint-disable max-len -->
  <path d="M282.3 343.7L248.1 376.1C244.5 381.5 238.4 384 232 384H192V424C192 437.3 181.3 448 168 448H128V488C128 501.3 117.3 512 104 512H24C10.75 512 0 501.3 0 488V408C0 401.6 2.529 395.5 7.029 391L168.3 229.7C162.9 212.8 160 194.7 160 176C160 78.8 238.8 0 336 0C433.2 0 512 78.8 512 176C512 273.2 433.2 352 336 352C317.3 352 299.2 349.1 282.3 343.7zM376 176C398.1 176 416 158.1 416 136C416 113.9 398.1 96 376 96C353.9 96 336 113.9 336 136C336 158.1 353.9 176 376 176z" /><!--  eslint-enable max-len -->
</svg>
`;

exports[`Icons > should render 'SettingsIcon' 1`] = `
<svg fill="#FFF" height="30" style="enable-background: new 0 0 24 24;" viewBox="0 0 24 24" width="30" xml:space="preserve" xmlns="http://www.w3.org/2000/svg">
  <g id="info" />
  <g id="icons">
    <!--  eslint-disable max-len -->
    <path d="M22.2,14.4L21,13.7c-1.3-0.8-1.3-2.7,0-3.5l1.2-0.7c1-0.6,1.3-1.8,0.7-2.7l-1-1.7c-0.6-1-1.8-1.3-2.7-0.7   L18,5.1c-1.3,0.8-3-0.2-3-1.7V2c0-1.1-0.9-2-2-2h-2C9.9,0,9,0.9,9,2v1.3c0,1.5-1.7,2.5-3,1.7L4.8,4.4c-1-0.6-2.2-0.2-2.7,0.7   l-1,1.7C0.6,7.8,0.9,9,1.8,9.6L3,10.3C4.3,11,4.3,13,3,13.7l-1.2,0.7c-1,0.6-1.3,1.8-0.7,2.7l1,1.7c0.6,1,1.8,1.3,2.7,0.7L6,18.9   c1.3-0.8,3,0.2,3,1.7V22c0,1.1,0.9,2,2,2h2c1.1,0,2-0.9,2-2v-1.3c0-1.5,1.7-2.5,3-1.7l1.2,0.7c1,0.6,2.2,0.2,2.7-0.7l1-1.7   C23.4,16.2,23.1,15,22.2,14.4z M12,16c-2.2,0-4-1.8-4-4c0-2.2,1.8-4,4-4s4,1.8,4,4C16,14.2,14.2,16,12,16z" id="settings" /><!--  eslint-enable max-len -->
  </g>
</svg>
`;

exports[`Icons > should render 'WorkflowJobsIcon' 1`] = `
<svg fill="#FFF" height="30" style="enable-background: new 0 0 256 256;" viewBox="0 0 24 24" width="30" xml:space="preserve" xmlns="http://www.w3.org/2000/svg">
  <!--  eslint-disable max-len -->
  <path d="M10.984 13.836a.5.5 0 0 1-.353-.146l-.745-.743a.5.5 0 1 1 .706-.708l.392.391 1.181-1.18a.5.5 0 0 1 .708.707l-1.535 1.533a.504.504 0 0 1-.354.146zm9.353-.147l1.534-1.532a.5.5 0 0 0-.707-.707l-1.181 1.18-.392-.391a.5.5 0 1 0-.706.708l.746.743a.497.497 0 0 0 .706-.001zM4.527 7.452l2.557-1.585A1 1 0 0 0 7.09 4.17L4.533 2.56A1 1 0 0 0 3 3.406v3.196a1.001 1.001 0 0 0 1.527.85zm2.03-2.436L4 6.602V3.406l2.557 1.61zM24 12.5c0 1.93-1.57 3.5-3.5 3.5a3.503 3.503 0 0 1-3.46-3h-2.08a3.503 3.503 0 0 1-3.46 3 3.502 3.502 0 0 1-3.46-3h-.558c-.972 0-1.85-.399-2.482-1.042V17c0 1.654 1.346 3 3 3h.04c.244-1.693 1.7-3 3.46-3 1.93 0 3.5 1.57 3.5 3.5S13.43 24 11.5 24a3.502 3.502 0 0 1-3.46-3H8c-2.206 0-4-1.794-4-4V9.899A5.008 5.008 0 0 1 0 5c0-2.757 2.243-5 5-5s5 2.243 5 5a5.005 5.005 0 0 1-4.952 4.998A2.482 2.482 0 0 0 7.482 12h.558c.244-1.693 1.7-3 3.46-3a3.502 3.502 0 0 1 3.46 3h2.08a3.503 3.503 0 0 1 3.46-3c1.93 0 3.5 1.57 3.5 3.5zm-15 8c0 1.378 1.122 2.5 2.5 2.5s2.5-1.122 2.5-2.5-1.122-2.5-2.5-2.5S9 19.122 9 20.5zM5 9c2.206 0 4-1.794 4-4S7.206 1 5 1 1 2.794 1 5s1.794 4 4 4zm9 3.5c0-1.378-1.122-2.5-2.5-2.5S9 11.122 9 12.5s1.122 2.5 2.5 2.5 2.5-1.122 2.5-2.5zm9 0c0-1.378-1.122-2.5-2.5-2.5S18 11.122 18 12.5s1.122 2.5 2.5 2.5 2.5-1.122 2.5-2.5zm-13 8a.5.5 0 1 0 1 0 .5.5 0 0 0-1 0zm2 0a.5.5 0 1 0 1 0 .5.5 0 0 0-1 0zm12 0c0 1.93-1.57 3.5-3.5 3.5a3.503 3.503 0 0 1-3.46-3.002c-.007.001-.013.005-.021.005l-.506.017h-.017a.5.5 0 0 1-.016-.999l.506-.017c.018-.002.035.006.052.007A3.503 3.503 0 0 1 20.5 17c1.93 0 3.5 1.57 3.5 3.5zm-1 0c0-1.378-1.122-2.5-2.5-2.5S18 19.122 18 20.5s1.122 2.5 2.5 2.5 2.5-1.122 2.5-2.5z" stroke-width="0" /><!--  eslint-enable max-len -->
</svg>
`;
